### PR TITLE
[CSharpBinding] Fixed failing unit test.

### PR DESCRIPTION
**Pull request by mkrueger**

### CODE REVIEW

**Comment by mkrueger** [comment]
This solution prevents to create a cached OptionsSet object and queries the "real" options every time. Should even use less memory for translating roslyn options from ours.

File: main/src/addins/CSharpBinding/MonoDevelop.CSharp.OptionProvider/CSharpDocumentOptionsProvider.cs [hunk 1]
```diff
@@ -87,12 +76,20 @@ static string GetPath(Document document)
 
 		internal class DocumentOptions : IDocumentOptions
 		{
-			readonly OptionSet optionSet;
+			readonly static IEnumerable<string> types = Ide.IdeServices.DesktopService.GetMimeTypeInheritanceChain (CSharpFormatter.MimeType);
```
Comment (Therzok) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp.OptionProvider/CSharpDocumentOptionsProvider.cs]: Can this change at runtime?

Comment (mkrueger) [reply main/src/addins/CSharpBinding/MonoDevelop.CSharp.OptionProvider/CSharpDocumentOptionsProvider.cs]: No

**Comment by slluis** [comment]
I cherrypicked the fix into new-doc-model.

**Pull request closed without merging**